runtime/lua/vim/lsp/buf.lua

```diff
@@ -296,11 +296,23 @@ function M.rename(new_name)
       --
       -- see https://microsoft.github.io/language-server-protocol/specification#textDocument_prepareRename
       opts.default = vfn.expand('<cword>')
-      if not new_name then npcall(vim.ui.input, opts, on_confirm) end
+      if not new_name then npcall(vim.ui.input, opts, request_rename) end
     end
-    if new_name then on_confirm(new_name) end
+    if new_name then request_rename(new_name) end
+  end
+
+  -- Only send prepareRename request if all servers attached to the buffer support it
+  local supports_prepare_rename = true
+  for _, client in pairs(vim.lsp.buf_get_clients()) do
+    supports_prepare_rename = client.supports_method('textDocument/prepareRename')
+  end
+
+  if supports_prepare_rename then
+    request('textDocument/prepareRename', util.make_position_params(), prepare_rename)
+  else
+    new_name = new_name or npcall(vfn.input, "New Name: ", vfn.expand('<cword>'))
```

Comment:
Any reason to not use `vim.ui.input`? Since it's used other places for renaming inside `request_rename`

Reply:
```lua
    if new_name then
      request_rename(new_name)
    else
      npcall(vim.ui.input, { prompt= "New Name: ", default = vfn.expand('<cword>')}, request_rename)
    end
```
Worked for me.